src/ssz/tree_view/list_composite.zig

```diff
@@ -0,0 +1,208 @@
+const CompositeChunks = @import("chunks.zig").CompositeChunks;
+
+/// A specialized tree view for SSZ list types with composite element types.
+/// Each element occupies its own subtree.
+pub fn ListCompositeTreeView(comptime ST: type) type {
+    comptime {
+        if (ST.kind != .list) {
+            @compileError("ListCompositeTreeView can only be used with List types");
+        }
+        if (!@hasDecl(ST, "Element") or isBasicType(ST.Element)) {
+            @compileError("ListCompositeTreeView can only be used with List of composite element types");
+        }
+    }
+
+    return struct {
+        base_view: BaseTreeView,
+
+        pub const SszType = ST;
+        pub const Element = ST.Element.TreeView;
+
+        const Self = @This();
+
+        const base_chunk_depth: Depth = @intCast(ST.chunk_depth);
+        const chunk_depth: Depth = chunkDepth(Depth, base_chunk_depth, ST);
+        const list_length_gindex: Gindex = Gindex.fromDepth(1, 1);
+        const Chunks = CompositeChunks(ST, chunk_depth);
+
+        pub fn init(allocator: Allocator, pool: *Node.Pool, root: Node.Id) !Self {
+            return Self{
+                .base_view = try BaseTreeView.init(allocator, pool, root),
+            };
+        }
+
+        pub fn deinit(self: *Self) void {
+            self.base_view.deinit();
+        }
+
+        pub fn commit(self: *Self) !void {
+            if (self.base_view.data.changed.contains(list_length_gindex)) {
+                if (self.base_view.data.list_length) |len| {
+                    const length_node = try self.base_view.pool.createLeafFromUint(len);
+                    const opt_old = blk: {
+                        errdefer self.base_view.pool.unref(length_node);
+                        break :blk try self.base_view.data.children_nodes.fetchPut(
+                            self.base_view.allocator,
+                            list_length_gindex,
+                            length_node,
+                        );
+                    };
+                    if (opt_old) |old_entry| {
+                        if (old_entry.value.getState(self.base_view.pool).getRefCount() == 0) {
+                            self.base_view.pool.unref(old_entry.value);
+                        }
+                    }
+                }
+            }
+
+            try self.base_view.commit();
+        }
+
+        pub fn clearCache(self: *Self) void {
+            self.base_view.data.clearChildrenNodesCache(self.base_view.pool);
+            self.base_view.data.clearChildrenDataCache(self.base_view.allocator, self.base_view.pool);
+            self.base_view.data.changed.clearRetainingCapacity();
+            self.base_view.data.list_length = null;
+        }
+
+        pub fn hashTreeRoot(self: *Self, out: *[32]u8) !void {
+            try self.commit();
+            out.* = self.base_view.data.root.getRoot(self.base_view.pool).*;
+        }
+
+        pub fn getLength(self: *Self) !usize {
```

Comment:
this is a bit verbose because we never have `setLength()`
maybe either `length()` or `len()`is enough and more compatible to the regular type
don't have strong opinion here but let's see how @spiral-ladder thinks